device/globals/global_vm.gd

```diff
@@ -370,14 +370,17 @@ func register_object(name, val):
 		report_errors("register_object", ["Name not given for " + val.get_class()])
 
 	objects[name] = val
-	if name in states:
-		val.set_state(states[name])
-	else:
-		val.set_state("default")
-	if name in actives:
-		val.set_active(actives[name])
 	val.connect("tree_exited", self, "object_exit_scene", [name])
 
+	# Most objects have states/animations, but targets do not
+	if not val is esc_type.TARGET:
```

Comment:
Isn't it better to stub `set_state` in `target.gd`, as was the case for `player.gd` until recently, in case we want to implement it in the future? I can't think of a reasonable implementation of state for a `Position2D` though.

Reply:
As per the comment below, there really isn't a need.

Reply:
I remembered seeing `has_method()` so I used that instead of having those types. Quacks like a duck and all that...